packages/docs/build/pug-to-json.js

```diff
@@ -0,0 +1,80 @@
+"use strict"
+const pug = require('pug')
+const walk = require('pug-walk')
+const compiler = require('vue-template-compiler')
+
+module.exports = function (content) {
+  this.cacheable()
+
+  const page = { children: [] }
+  let pageSection
+
+  function startSection () {
+    pageSection = {
+      type: 'section',
+      children: [],
+    }
+    page.children.push(pageSection)
+  }
+
+  pug.render(content, {
+    compileDebug: false,
+    filename: this.resourcePath,
+    filters: {
+      code: text => text
+    },
+    plugins: [{
+      preFilters (ast) {
+        const parents = []
+        walk(ast, node => {
+          if (!pageSection || ['h2', 'up-next', 'usage', 'usage-new', 'api', 'examples'].includes(node.name)) {
+            startSection()
+          }
+          switch (node.type) {
+            case 'Tag':
+              if (node.name === 'h2') {
+                pageSection.children.push({
+                  type: 'heading',
+                  lang: node.block.nodes[0].val
+                })
+              } else if (node.name === 'code') {
+                const lang = eval(node.attrs.find(attr => attr.name === 'lang').val)
+                const src = eval(node.attrs.find(attr => attr.name === 'src').val)
+                // TODO: inline code
```

Comment:
We could keep the edit button and even have it link to the correct line by using `node.line`